core/Pos/Core/Class.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleInstances #-}
 
```

Comment:
That seems to be the only change to that file. Is it really necessary?

Oh, maybe its listed in the cabal file, and needed in the file so you can load this into GHCi.

Reply:
yep exactly, default-extensions makes it imposible to develop anything without having cabal hand-hold you the entire way

Reply:
This problem is caused by having `.` in the `source-dirs` in the cabal stanza. Removing that will let the default-extensions do their thing.

Reply:
`default-extensions` doesnt work if you bypass cabal and directly use tools like `ghcid`